km62/Pochta_Ivan/laboratory1.sql

```diff
@@ -1,2 +1,182 @@
 -- LABORATORY WORK 1
 -- BY Pochta_Ivan
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+модифікувати таблиці та вставляти дані
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE USER pochta IDENTIFIED BY johnp
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+Alter user pochta QUota 100M on "USERS"; --додано в якості власного code review
+GRANT "CONNECT" TO pochta;
+GRANT ALTER ANY TABLE TO pochta;
+GRANT INSERT ANY TABLE TO pochta;
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент купляє квиток на потяг.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE TABLE STUDENT(
+  student_id varchar(5) not null,
+  student_name varchar(25) not null
+);
+ALTER TABLE STUDENT ADD CONSTRAINT student_pk PRIMARY KEY( student_id );
+
+CREATE TABLE TICKET(
+  ticket_id varchar(5) not null,
+  train_id varchar(5) not null,
+  customer_id varchar(5) not null
+);
+ALTER TABLE TICKET ADD CONSTRAINT ticket_pk PRIMARY KEY (ticket_id);
+CREATE TABLE STUDENT_BUY_TICKET(
+  student_fk varchar(5) not null,
+  ticket_fk varchar(5) not null 
+);
+ALTER TABLE STUDENT_BUY_TICKET ADD CONSTRAINT sbt_pk PRIMARY KEY (ticket_fk);
+ALTER TABLE STUDENT_BUY_TICKET  ADD CONSTRAINT sbt_student_fk FOREIGN KEY (student_fk) REFERENCES  STUDENT(student_id);
+ALTER TABLE STUDENT_BUY_TICKET  ADD CONSTRAINT sbt_ticket_fk FOREIGN KEY (ticket_fk) REFERENCES  TICKET(ticket_id);
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+GRANT SELECT ANY TABLE to POCHTA;
+GRANT ALTER ANY TABLE to pochta;
+GRANT Create ANY TABLE to POCHTA;
+GRANT INSERT ANY TABLE to pochta;
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Якого товару найменше продано?
+Виконати завдання в SQL. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+
+
+
+
```

Comment:
`SELECT
    prod_id,
    "sum"
FROM (
    SELECT
        orderitems.prod_id,
        SUM(orderitems.quantity) as "sum"
    FROM
        orderitems
    GROUP BY 
        orderitems.prod_id)
WHERE
    "sum" IN (SELECT MIN("sum") FROM (SELECT
        orderitems.prod_id,
        SUM(orderitems.quantity) as "sum"
    FROM
        orderitems
    GROUP BY 
        orderitems.prod_id));
`